modules/cluster-logging-collector-log-forward-cloudwatch.adoc

```diff
@@ -0,0 +1,294 @@
+
+You can forward logs to Amazon CloudWatch, a hosted monitoring and log storage service. You can forward logs to CloudWatch in addition to, or instead of, the default OpenShift Logging-managed Elasticsearch log store.
+
+To configure log forwarding to CloudWatch, create a `ClusterLogForwarder` custom resource (CR) with an output for CloudWatch and a pipeline that uses the output.
+
+.Procedure
+
+. Create a `Secret` YAML file similar to the following example. Use base64-encoded values for the `aws_access_key_id` and `aws_secret_access_key` fields. For example:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cw-secret
+  namespace: openshift-logging
+data:
+  aws_access_key_id: QUtJQUlPU0ZPRE5ON0VYQU1QTEUK
+  aws_secret_access_key: d0phbHJYVXRuRkVNSS9LN01ERU5HL2JQeFJmaUNZRVhBTVBMRUtFWQo=
+----
+
+
+. Create the secret. For example:
++
+[source,terminal]
+----
+$ oc apply -f cw-secret.yaml
+----
+
+
+. Create a `ClusterLogForwarder` CR YAML file. In it, specify the name of the secret. For example:
++
+[source,yaml]
+----
+apiVersion: "logging.openshift.io/v1"
+kind: ClusterLogForwarder
+metadata:
+  name: instance <1>
+  namespace: openshift-logging <2>
+spec:
+  outputs:
+   - name: cw <3>
+     type: cloudwatch <4>
+     cloudwatch:
+       groupBy: logType <5>
+       region: us-east-2 <6>
+     secret:
+        name: cw-secret <7>
+  pipelines:
+    - name: infra-logs <8>
+      inputRefs: <9>
+        - infrastructure
+        - audit
+        - application
+      outputRefs:
+        - cw <10>
+----
+<1> The name of the `ClusterLogForwarder` CR must be `instance`.
+<2> The namespace for the `ClusterLogForwarder` CR must be `openshift-logging`.
+<3> Specify a name for the output.
+<4> Specify the `cloudwatch` type.
+<5> Optional: Specify the option CloudWatch uses to group the logs. If you omit `groupBy`, its default value is `logType`.
+<6> Specify the region your CloudWatch instance uses.
+<7> Specify the name of the secret.
+<8> Optional: Specify a name for the pipeline.
+<9> Specify which log types to forward by using the pipeline: `application,` `infrastructure`, or `audit`.
+<10> Specify the name of the output to use when forwarding logs with this pipeline.
+
+. Create the CR object:
++
+[source,terminal]
+----
+$ oc create -f <file-name>.yaml
+----
+
+The Red Hat OpenShift Logging Operator redeploys the Fluentd pods. If the pods do not redeploy, you can delete the Fluentd pods to force them to redeploy.
+
+[source,terminal]
+----
+$ oc delete pod --selector logging-infra=fluentd
+----
+
+'''
+
+.Example: How various configurations of the ClusterLogForwarder affect CloudWatch
+
+Suppose that you run an {product-title} cluster on AWS named `mycluster` and the value of its `infrastructureName` is `mycluster-7977k`.
+
+[source,terminal]
+----
+$ oc get Infrastructure/cluster -ojson | jq .status.infrastructureName
+"mycluster-7977k"
+----
+
+You can see the name of the default project `oc` is using.
```

Comment:
The fact that 'app' is the default project is not quite the main point. It just saves you from typing "oc -n app" all the time. The main point is that there's a project 'app' that user created for his application, and there's an application pod running in it, so that we have a source of logs of type "application" for the example.

Reply:
Should I simply delete lines 101 - 108?

Reply:
Deleted.